### PR TITLE
Add partial support for ipset

### PR DESCRIPTION
**Pull request by corny**
Looking forward to read your comments and improve my code.

### CODE REVIEW

File: ipset_linux.go [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+import (
+	"log"
+	"net"
+	"syscall"
+
+	"github.com/vishvananda/netlink/nl"
+	"golang.org/x/sys/unix"
+)
+
+type IPSetEntry struct {
```
Comment (jellonek) [inline ipset_linux.go]: Exported name without docstring.
Please add docstrings to all names which are started with upper case, using convention described in https://blog.golang.org/godoc-documenting-go-code

File: ipset_linux.go [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+		case nl.IPSET_ATTR_FAMILY:
+			result.Family = attr.Value[0]
+		case nl.IPSET_ATTR_FLAGS:
+			result.Flags = attr.Value[0]
+		case nl.IPSET_ATTR_DATA | nl.NLA_F_NESTED:
+			result.parseAttrData(attr.Value)
+		case nl.IPSET_ATTR_ADT | nl.NLA_F_NESTED:
+			result.parseAttrADT(attr.Value)
+		default:
+			log.Println("unknown type", attr.Type)
```
Comment (aboch) [inline ipset_linux.go]: Please add more context to this message: log.Printf + `unknown ipset message attribute from kernel: %v` ?

Comment (corny) [reply ipset_linux.go]: Good idead, done.

**Comment by vishvananda** [comment]
This is looking good. Can you please squash it down into a single commit?

File: ipset_linux.go (hunk 1 above)
Comment (corny) [reply ipset_linux.go]: done

**Comment by aboch** [comment]
You may want to remove these two binaries from the diffset

image

**Comment by corny** [comment]
Is there anything missing in my pull request?

**Comment by corny** [comment]
> You may want to remove these two binaries from the diffset

These are netlink responses. They are required by `ipset_linux_test.go`.

**Comment by onokonem** [comment]
hi @corny 

I have a problem with this patch

```
$uname -a
Darwin AF6438.local 18.2.0 Darwin Kernel Version 18.2.0: Fri Oct  5 19:41:49 PDT 2018; root:xnu-4903.221.2~2/RELEASE_X86_64 x86_64

$go version
go version go1.11.2 darwin/amd64

$GOOS=linux go build ./netmanager/cmd/netmanager
# github.com/AnchorFreePartner/services/vendor/github.com/vishvananda/netlink
../../vendor/github.com/vishvananda/netlink/ipset_linux.go:211:34: undefined: unix.NFNL_SUBSYS_IPSET
```

any idea why?

**Comment by corny** [comment]
@onokonem You don't have the proper header files on your system.

**Comment by corny** [comment]
@vishvananda ping

**Comment by vishvananda** [comment]
@corny  Thanks for updating this. Our policy is for prs to be a single commit. Can you squash this down? You can add a co-authored-by line for the original author so we don't lose their contributions.

**Comment by corny** [comment]
I squashed it

**Pull request merged**